### DISCUSSION TITLE
diff-mode keymap conflict with nvim-treesitter

### DISCUSSION THREAD

**Discussion by krishna-bala** [LazyVim/LazyVim#1588] [Q&A] [answered]
A recent [change](https://github.com/LazyVim/LazyVim/commit/0e66ef83934eb62ee2237e331dcbc0a22b809aae#diff-4b4b63b67b4dbab286cc1411b968e601c2078ca33f404e9453496f6ba73c3af4) to [treesitter.lua](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/treesitter.lua) overwrites some default mappings in `diff-mode`. Specifically, the `diff-mode` mappings `[c`, `[C`, `]c`, and `]C` were overwritten by the mappings for treesitter.

I like the new treesitter mappings and was wondering how I could remap the `diff-mode` mappings for `jumpto-diffs` to something other than `[c`, e.g. `[h`. However, I'm not sure how to do that. The `diff` [help file](https://neovim.io/doc/user/diff.html#jumpto-diffs) doesn't explain the mechanism for `[c`.

I found a way to determine if the current window is in `diff-mode` [here](https://www.reddit.com/r/neovim/comments/ot4dvr/in_lua_how_can_i_query_diff_and_then_do_stuff/):
```
if vim.api.nvim_win_get_option(0, "diff") then
    -- do whatever
end
```

I was thinking I could make use of the `expr` function for vim keymaps, something along the lines of
```
vim.keymap.set('n`, '[c', function()
  if vim.api.nvim_win_get_option(0, "diff") then 
    return '...' 
  end
  return '...'
end, { expr = true })
```
but I can't seem to figure out (1) the command to execute the `jumpto-diffs` movement and (2) the command to execute a `textobjects` movement.

This is probably the wrong approach, but wanted to try something before posting. Any help or ideas appreciated.

**Comment by krishna-bala** [comment]
For context, the motivation to find a different keymap is because I use nvim as my `git difftool` command from the command line.

**Reply by folke** [reply, marked as answer]
I just pushed a fix where we simply delete the treesitter keymaps when a buffer loads when in diff mode.
